packages/web/src/components/gcds-topic-menu/gcds-topic-menu.tsx

```diff
@@ -0,0 +1,409 @@
+  * Toggle open theme and topic menu
+  */
+  @Method()
+  async toggleNav() {
+    this.open = !this.open;
+
+    if (this.open) {
+      if (this.navSize == 'desktop') {
+        this.themeList.children[0].children[0].setAttribute('aria-expanded', 'true');
+      } else {
+        // Close most requested on mobile
+        this.el.shadowRoot.querySelectorAll('[data-keep-expanded]').forEach((el) => {
+          el.setAttribute('aria-expanded', 'false');
+        });
+      }
+
+      setTimeout(() => {
+        this.themeList.children[0].querySelector('a').focus();
+      }, 50);
+
+      await this.updateNavItemQueue(this.themeList);
+    } else {
+      this.closeAllMenus();
+    }
+  }
+
+  /*
+  * Pass new window size: desktop or mobile
+  */
+  @Method()
+  async updateNavSize(size) {
+    this.navSize = size;
+  }
+
+  /*
+  * Get current navSize state
+  */
+  @Method()
+  async getNavSize() {
+    return this.navSize;
+  }
+
+  /**
+  * Update keyboard focus queue
+  */
+  @Method()
+  async updateNavItemQueue(parent) {
+    let focusableElements = [];
+
+    for (let x = 0; x < parent.children.length; x++) {
+      let link = parent.children[x].querySelector('a');
+      if (link) {
+        focusableElements.push(link);
+
+        if (link.hasAttribute('data-keep-expanded') && this.navSize == 'desktop') {
+          for (let c = 0; c < link.parentNode.children[1].children.length; c++) {
+            focusableElements.push(link.parentNode.children[1].children[c].querySelector('a'));
+          }
+        }
+      }
+    }
+
+    this.navItems = focusableElements;
+  }
+
+  /**
+  * Focus menu link
+  */
+  private focusMenuLink(queue, activeElement, nextStep) {
+    if (activeElement.closest('ul').hasAttribute('data-top-menu') && activeElement.hasAttribute('aria-haspopup') && !activeElement.hasAttribute('data-keep-expanded')) {
+      this.closeAllMenus();
+    } else if (activeElement.hasAttribute('aria-haspopup') && !activeElement.hasAttribute('data-keep-expanded')) {
+      activeElement.setAttribute('aria-expanded', 'false');
+    }
+
+    queue[nextStep].focus();
+
+    if (queue[nextStep].hasAttribute('aria-haspopup') && this.navSize == 'desktop') {
+      queue[nextStep].setAttribute('aria-expanded', 'true');
+    }
+  }
+
+  /*
+  * Observe lang attribute change
+  */
+  updateLang() {
+    const observer = new MutationObserver((mutations) => {
+      if (mutations[0].oldValue != this.el.lang) {
+        this.lang = this.el.lang;
+      }
+    });
+    observer.observe(this.el, observerConfig);
+  }
+
+  async componentWillLoad() {
+    // Define lang attribute
+    this.lang = assignLanguage(this.el);
+
+    this.updateLang();
+
+    const mediaQuery = window.matchMedia('screen and (max-width: 991px)');
+
+    if (mediaQuery.matches) {
+      this.navSize = 'mobile';
+    } else {
+      this.navSize = 'desktop';
+    }
+
+    return await fetch(`https://www.canada.ca/content/dam/canada/sitemenu/sitemenu-v2-${this.lang}.html`)
```

Comment:
Could we use a try...catch here instead to handle any potential fetch failures that might occur?